tmt/hardware.py

```diff
@@ -1080,6 +1080,30 @@ def _parse_disks(spec: Spec) -> BaseConstraint:
     return group
 
 
+@ungroupify
+def _parse_bootdisks(spec: Spec) -> BaseConstraint:
+    """
+    Parse bootdisk-related constraints.
+
+    :param spec: raw constraint block specification.
+    :returns: block representation as :py:class:`ConstraintBase` or one of its subclasses.
+    """
+
+    group = And()
+    group.constraints += [
+        TextConstraint.from_specification(
+            'boot-disk',
+            bootdisk_spec,
+            allowed_operators=[Operator.EQ, Operator.NEQ, Operator.MATCH, Operator.NOTMATCH])
+        for _, bootdisk_spec in enumerate(spec)
+        ]
+
+    # Do not reduce group to a single item: we need more predictable tree
+    # for Beaker driver to be able to construct its own XML.
```

Comment:
Hmm, this is weird, no other constraint seems to require this. Plus, the simplification happens thanks to the `@ungroupify` decorator above, so this comment describes something that is not happening.